hal/common/retarget.cpp

```diff
@@ -72,6 +72,21 @@ extern const char __stdout_name[] = "/stdout";
 extern const char __stderr_name[] = "/stderr";
 #endif
 
+#if defined MBED_CFG_DEBUG_OPTIONS_COVERAGE
+/* Coverage sync flag used between greentea_client\source\test_env.cpp:greentea_notify_completion()
+ * and _open() to indicate that file is being open by gcov.
+ */
+bool coverage_report = false;
```

Comment:
Doesn't `greentea` also need a change to work with this flag? Or is this covered in a separate PR (or maybe it already does)?